### PR TITLE
Config parser refactoring

### PR DESCRIPTION
**Pull request by pgdr**
**Task**
Give `config_parser.py` some Pythonic love.

**Approach**
Add `__getitem__` and `__len__` and tests.   Fixes a broken (probably unused?) method `getSchemaItem` (it never returned anything).  Fixes whitespace.

**Pre un-WIP checklist**
- [X] Statoil tests pass locally
- [X] Have completed graphical integration test steps

### CODE REVIEW

File: python/tests/res/config/test_config.py [hunk 1]
```diff
@@ -66,8 +66,8 @@ def test_item_types(self):
             schema_item.iset_type(2 , ContentTypeEnum.CONFIG_BOOL )
             schema_item.iset_type(3 , ContentTypeEnum.CONFIG_STRING )
             schema_item.iset_type(4 , ContentTypeEnum.CONFIG_PATH )
-            self.assertFalse( "TYPE_XX" in conf )
-            self.assertTrue( "TYPE_ITEM" in conf )
+            self.assertNotIn( "TYPE_XX", conf )
```
Comment (joakim-hove) [inline python/tests/res/config/test_config.py]: Cool!

File: python/tests/res/config/test_config.py [hunk 2]
```diff
@@ -91,7 +91,10 @@ def test_item_types(self):
             self.assertEqual( path_value , "file")
             self.assertEqual( type_item.igetString(4) , "file")
 
-
+            # test __getitem__
+            self.assertTrue(conf['TYPE_ITEM'])
```
Comment (joakim-hove) [inline python/tests/res/config/test_config.py]: This is rather a test of the truthiness of the `ConfigSchemaItem`?

Comment (pgdr) [reply python/tests/res/config/test_config.py]: Yes, you could say that, but also that  `__getitem__` is implemented, and that it returns a truthy type for the existing key `TYPE_ITEM`.  Should I remove the comment?

**Review by joakim-hove — APPROVED** [review]

File: python/python/res/config/config_parser.py [hunk 3]
```diff
@@ -17,67 +17,86 @@
...
 
+    def add(self, keyword, required=False, value_type=None):
+        item = self._add(keyword, required).setParent(self)
+        if value_type:
+            item.iset_type(0, value_type)
         return item
 
 
-    def getSchemaItem(self , keyword):
+    def getSchemaItem(self, keyword):
```
Comment (markusdregi) [inline python/python/res/config/config_parser.py:59]: I think we should we deprecate this one..

File: python/tests/res/config/test_config.py (hunk 2 above)
Comment (joakim-hove) [reply python/tests/res/config/test_config.py]: Fine - leave it!

File: python/python/res/config/config_parser.py (hunk 3 above)
Comment (pgdr) [reply python/python/res/config/config_parser.py:59]: done, I will merge when green

**Comment by pgdr** [comment]
I have fixed downstream call sites post #66 and issued a rebuild.

See Statoil/ert#48.

**Pull request merged**